sdk/messaging/azservicebus/internal/mgmt.go

```diff
@@ -34,171 +31,39 @@ const (
 	DeferredDisposition  DispositionStatus = "defered"
 )
 
-type (
-	mgmtClient struct {
-		ns    NamespaceForMgmtClient
-		links AMQPLinks
-
-		clientMu sync.Mutex
-		rpcLink  RPCLink
-
-		sessionID          *string
-		isSessionFilterSet bool
-	}
-)
-
-type MgmtClient interface {
-	Close(ctx context.Context) error
-	SendDisposition(ctx context.Context, lockToken *amqp.UUID, state Disposition, propertiesToModify map[string]interface{}) error
-	ReceiveDeferred(ctx context.Context, mode ReceiveMode, sequenceNumbers []int64) ([]*amqp.Message, error)
-	PeekMessages(ctx context.Context, fromSequenceNumber int64, messageCount int32) ([]*amqp.Message, error)
-
-	ScheduleMessages(ctx context.Context, enqueueTime time.Time, messages ...*amqp.Message) ([]int64, error)
-	CancelScheduled(ctx context.Context, seq ...int64) error
-
-	RenewLocks(ctx context.Context, linkName string, lockTokens []amqp.UUID) ([]time.Time, error)
-	RenewSessionLock(ctx context.Context, sessionID string) (time.Time, error)
-
-	GetSessionState(ctx context.Context, sessionID string) ([]byte, error)
-	SetSessionState(ctx context.Context, sessionID string, state []byte) error
+type mgmtError struct {
```

Comment:
@jhendrixMSFT: this whole file went through some refactoring. Basically, the mgmtClient used to be a struct and it held onto an RPCLink, which it would protect using a lock. 

Looking at it more, it just seemed like overkill and I already had all this kind of locking code in amqpLinks.go. So I basically just made each function standalone here, and made it take an amqpLinks instead.

I also modified amqpLinks to return an RPCLink instead of a MgmtClient. This just makes things simpler since it means link management remains in amqplinks.go and isn't spread around anymore.